src/Microsoft.AspNetCore.Mvc.Formatters.Json/JsonInputFormatter.cs

```diff
@@ -98,6 +127,18 @@ public class JsonInputFormatter : TextInputFormatter
             }
 
             var request = context.HttpContext.Request;
+
+            if (_bufferRequestBody)
```

Comment:
We should get `request.Body.CanSeek` here. Then if the body is already buffered we don't need to do anything.